nucypher-core-python/nucypher_core/ferveo.pyi

```diff
@@ -0,0 +1,261 @@
+from typing import Sequence
+
+
+class Keypair:
+    @staticmethod
+    def random() -> Keypair:
+        ...
+
+    @staticmethod
+    def from_secure_randomness(data: bytes) -> Keypair:
+        ...
+
+    @staticmethod
+    def secure_randomness_size(data: bytes) -> int:
+        ...
+
+    @staticmethod
+    def from_bytes(data: bytes) -> Keypair:
+        ...
+
+    def __bytes__(self) -> bytes:
+        ...
+
+    def public_key(self) -> FerveoPublicKey:
+        ...
+
+
+class FerveoPublicKey:
+    @staticmethod
+    def from_bytes(data: bytes) -> FerveoPublicKey:
+        ...
+
+    def __bytes__(self) -> bytes:
+        ...
+
+    def __hash__(self) -> int:
+        ...
+
+
+class Validator:
+
+    def __init__(self, address: str, public_key: FerveoPublicKey):
+        ...
+
+    address: str
+
+    public_key: FerveoPublicKey
+
+
+class Transcript:
+    @staticmethod
+    def from_bytes(data: bytes) -> Transcript:
+        ...
+
+    def __bytes__(self) -> bytes:
+        ...
+
+
+class DkgPublicKey:
+    @staticmethod
+    def from_bytes(data: bytes) -> DkgPublicKey:
+        ...
+
+    def __bytes__(self) -> bytes:
+        ...
+
+
+class Dkg:
+
+    def __init__(
+            self,
+            tau: int,
+            shares_num: int,
+            security_threshold: int,
+            validators: Sequence[Validator],
+            me: Validator,
+    ):
+        ...
+
+    public_key: DkgPublicKey
+
+    public_params: DkgPublicParameters
+
+    def generate_transcript(self) -> Transcript:
+        ...
+
+    def aggregate_transcripts(self, messages: Sequence[(Validator, Transcript)]) -> AggregatedTranscript:
```

Comment:
Does `(Validator, Transcript)` work with type checkers? I would think you'd have to do `Tuple[Validator, Transcript]`

Reply:
(same in `AggregatedTranscript`)

Reply:
It works, but `Tuple[Validator, Transcript]` is better